### PR TITLE
Made pep8 compliant

### PR DESCRIPTION
**Pull request by nilshah98**
**PR Details**
- Used tool `autopep8` to automate basic pep8 rules
- Fixed line break issues
- Commit history contains all folder-wise commits

### CODE REVIEW

File: src/kicadtoNgspice/DeviceModel.py [hunk 1]
```diff
@@ -1,139 +1,164 @@
...
+                            # child.tag[1],eachline[0],eachline[1]
                             try:
-                                if os.path.exists(json_data["deviceModel"][key][0]):
-                                    self.entry_var[self.count].setText(json_data["deviceModel"][key][0])
-                                    path_name = json_data["deviceModel"][key][0]
+                                if os.path.exists(
+                                        json_data["deviceModel"][key][0]):
+                                    self.entry_var[self.count].setText(
+                                        json_data["deviceModel"][key][0])
+                                    path_name = json_data
```
Comment (sunilshetye) [inline src/kicadtoNgspice/DeviceModel.py]: need backslash at the end

File: src/kicadtoNgspice/DeviceModel.py [hunk 2]
```diff
@@ -1,139 +1,164 @@
...
-                            #print "DEVICE MODEL MATCHING---",child.tag[0],child.tag[1],eachline[0],eachline[1]
+                            # print "DEVICE MODEL MATCHING---",child.tag[0],\
+                            # child.tag[1],eachline[0],eachline[1]
                             try:
-                                if os.path.exists(json_data["deviceModel"][key][0]):
-                                    path_name = json_data["deviceModel"][key][0]
-                                    self.entry_var[self.count].setText(json_data["deviceModel"][key][0])
+                                if os.path.exists(
+                                        json_data["deviceModel"][key][0]):
+                                    path_name = json_data
```
Comment (sunilshetye) [inline src/kicadtoNgspice/DeviceModel.py]: need backslash at the end for continuation

File: src/kicadtoNgspice/DeviceModel.py [hunk 3]
```diff
@@ -144,50 +169,60 @@ def __init__(self, schematicInfo, clarg1):
...
+                            # child.tag[1],eachline[0],eachline[1]
                             try:
-                                if os.path.exists(json_data["deviceModel"][key][0]):
-                                    self.entry_var[self.count].setText(json_data["deviceModel"][key][0])
-                                    path_name = json_data["deviceModel"][key][0]
+                                if os.path.exists(
+                                        json_data["deviceModel"][key][0]):
+                                    self.entry_var[self.count].setText(
+                                        json_data["deviceModel"][key][0])
+                                    path_name = json_data
```
Comment (sunilshetye) [inline src/kicadtoNgspice/DeviceModel.py]: need backslash at the end for continuation

File: src/kicadtoNgspice/DeviceModel.py [hunk 4]
```diff
@@ -198,162 +233,187 @@ def __init__(self, schematicInfo, clarg1):
...
-                                self.entry_var[i].setText(json_data["deviceModel"][key][i-beg])
-                                if (i-beg) == 0:
-                                    if os.path.exists(json_data["deviceModel"][key][0]):
-                                        path_name = json_data["deviceModel"][key][0]
+                                self.entry_var[i].setText(
+                                    json_data["deviceModel"][key][i - beg])
+                                if (i - beg) == 0:
+                                    if os.path.exists(
+                                            json_data["deviceModel"][key][0]):
+                                        path_name = json_data
```
Comment (sunilshetye) [inline src/kicadtoNgspice/DeviceModel.py]: need backslash at the end for continuation

File: src/kicadtoNgspice/SubcircuitTab.py [hunk 5]
```diff
@@ -1,103 +1,115 @@
...
         self.grid = QtGui.QGridLayout()
         self.setLayout(self.grid)
-        
+
         for eachline in schematicInfo:
             words = eachline.split()
             if eachline[0] == 'x':
-                print "Subcircuit : Words",words[0]
-                self.obj_trac.subcircuitList[project_name+words[0]] = words
+                print(("Subcircuit : Words", words[0]))
```
Comment (sunilshetye) [inline src/kicadtoNgspice/SubcircuitTab.py]: use single bracket

File: src/kicadtoNgspice/SubcircuitTab.py [hunk 6]
```diff
@@ -1,103 +1,115 @@
...
-                #eg. If the line is 'x1 4 0 3 ua741', there are 3 ports(4, 0 and 3).
-                self.numPorts.append(len(words)-2)
-                print "Number of ports of sub circuit : ",self.numPorts
+                self.addbtn.setObjectName("%d" % self.count)
+                # Send the number of ports specified with the given\
+                # subcircuit for verification.
+                # eg. If the line is 'x1 4 0 3 ua741', there are 3 ports(4, 0
+                # and 3).
+                self.numPorts.append(len(words) - 2)
+                print(("Number of ports of sub circuit : ", self.numPorts))
```
Comment (sunilshetye) [inline src/kicadtoNgspice/SubcircuitTab.py]: use single bracket

File: src/ngspicetoModelica/NgspicetoModelica.py [hunk 7]
```diff
@@ -33,75 +30,79 @@ def readNetlist(self,filename):
                 f = open(filename)
             except Exception as e:
                 print("Error in opening file")
-                print(str(e))
+                print((str(e)))
```
Comment (sunilshetye) [inline src/ngspicetoModelica/NgspicetoModelica.py]: use single bracket

File: src/kicadtoNgspice/DeviceModel.py (hunk 4 above)
Comment (nilshah98) [reply src/kicadtoNgspice/DeviceModel.py]: Sir, it works properly without backslash as well, tested it in a mock file as -
```
import os
if os.path.exists(
    "."):
    print("works")
```

The above code executes successfully.
Is backslash to be added just as a syntactic sugar?

File: src/ngspicetoModelica/NgspicetoModelica.py (hunk 7 above)
Comment (nilshah98) [reply src/ngspicetoModelica/NgspicetoModelica.py]: Resolved under the documentation PR.

Should, I squash my commits, before you merge them for the documentation PR?

File: src/frontEnd/ProjectExplorer.py [hunk 8]
```diff
@@ -9,139 +9,181 @@ def __init__(self):
...
             self.textwindow.show()
         else:
-            self.obj_appconfig.current_project["ProjectName"]= str(self.filePath)
-            self.obj_appconfig.proc_dict[self.obj_appconfig.current_project['ProjectName']] = []
-            if self.obj_appconfig.current_project['ProjectName'] not in self.obj_appconfig.dock_dict:
-                self.obj_appconfig.dock_dict[self.obj_appconfig.current_project['ProjectName']] = []
-        
+            self.obj_appconfig.current_project["ProjectName"] = str(
+                self.filePath)
+            self.obj_appconfig.proc_dict
```
Comment (sunilshetye) [inline src/frontEnd/ProjectExplorer.py]: put backslash at the end.

File: src/kicadtoNgspice/DeviceModel.py (hunk 4 above)
Comment (sunilshetye) [reply src/kicadtoNgspice/DeviceModel.py]: Problem is not in the if block, but the line after that. It should be:

```
path_name = json_data\
      ["deviceModel"][key][0]
```

File: src/ngspicetoModelica/NgspicetoModelica.py (hunk 7 above)
Comment (sunilshetye) [reply src/ngspicetoModelica/NgspicetoModelica.py]: no, push a new commit in the documentation branch.

File: src/kicadtoNgspice/DeviceModel.py (hunk 4 above)
Comment (sunilshetye) [reply src/kicadtoNgspice/DeviceModel.py]: different ways to do this (anything is okay for me):

```
path_name = json_data\
      ["deviceModel"][key][0]

```

```
path_name = (json_data
      ["deviceModel"][key][0])

```

```
path_name = json_data[
      "deviceModel"][key][0]

```

Comment (nilshah98) [reply src/kicadtoNgspice/DeviceModel.py]: Thanks,
Resolved with the latest commit on documentation branch.

Ref - #86

File: src/frontEnd/ProjectExplorer.py (hunk 8 above)
Comment (nilshah98) [reply src/frontEnd/ProjectExplorer.py]: Resolved in the documentation PR - #86

**Pull request merged**